libminifi/src/Properties.cpp

```diff
@@ -122,26 +115,27 @@ bool Properties::parseConfigureFileLine(char *buf, std::string &prop_key, std::s
 
 // Load Configure File
 void Properties::loadConfigureFile(const char *fileName) {
-  std::string adjustedFilename;
-  if (fileName) {
-    // perform a naive determination if this is a relative path
-    if (fileName[0] != FILE_SEPARATOR_C) {
-      adjustedFilename = adjustedFilename + getHome() + FILE_SEPARATOR_C + fileName;
-    } else {
-      if (adjustedFilename.empty()) {
-        adjustedFilename = getHome();
-      }
-      adjustedFilename += fileName;
-    }
+  if (NULL == fileName) {
+    logger_->log_error("Configuration file path for %s is a nullptr!", getName().c_str());
+    return;
+  }
+
+  std::string adjustedFilename = getHome();
+  // perform a naive determination if this is a relative path
+  if (fileName[0] != utils::file::FileUtils::get_separator()) {
+    adjustedFilename += utils::file::FileUtils::get_separator();
   }
+
+  adjustedFilename += fileName;
+
   char *path = NULL;
 #ifndef WIN32
   char full_path[PATH_MAX];
   path = realpath(adjustedFilename.c_str(), full_path);
 #else
-  path = const_cast<char*>(adjustedFilename.c_str());
```

Comment:
was this here to avoid a warning?

Reply:
Yep, we assigned this to a simple char ptr, but that's now const char ptr.